dash.el

```diff
@@ -1591,6 +1591,54 @@ and when that result is non-nil, through the next form, etc."
                    (--> ,result ,form))
                  ,@more))))
 
+(defmacro -cond-> (x &rest clauses)
+  "Conditionally thread X through CLAUSES.
+Threads x (via `->') through each form for which the
+corresponding test expression is true.  Note that, unlike cond
+branching, `-cond->' threading does not short circuit after the
+first true test expression."
+  (declare (debug (form body))
+           (indent 1))
+  (when (-> clauses length (% 2) (= 1))
+    (error "Wrong number of arguments"))
+  (-let* ((g (dash--match-make-source-symbol))
```

Comment:
Wouldn't a simple `make-symbol` with a more descriptive name do? We don't strictly need monotonically increasing symbol names since this is a single binding.

Reply:
Perhaps. I was sure using a static symbol would expose it, bound to the the value being passed between expressions.
(This behaviour is clearly undesired, but I wished for it in the first versions. It made sense to test the current value, not just a snapshot of the first one. This is why i looked for `intern` (and became more confused).)
A quick test made me realise I was wrong. It will be corrected soon!

Reply:
`make-symbol` always creates a new symbol, even if it has the same name, they are in a different "intern table":

    (eq (make-symbol "a") (make-symbol "a"))  ;; => nil

`intern` puts the symbol in the global intern table *if it doesn't exist*, otherwise returns an existing symbol:

    (eq (intern "a") (intern "a"))  ;; => t

So that's it.  If you need a unique symbol which doesn't exist anywhere in emacs, use `make-symbol`, if you need to refer to a symbol which could've possible been created elsewhere, use `intern`:

    (funcall (intern "concat") "a" "b")  ;; => "ab", we refer to the concat function built-in in emacs
    (funcall (make-symbol "concat") "a" "b") ;; => error (void-function concat) because this symbol is looked up in it's own "universe", not the global one.